pulp_2to3_migration/app/migration.py

```diff
@@ -166,6 +167,20 @@ def complex_repo_migration(plugin_type, pulp3_repo_setup, repo_name):
     distributor_types = list(distributor_migrators.keys())
     repo_versions_setup = pulp3_repo_setup[repo_name]['repository_versions']
 
+    signing_service = None
+    signing_service_name = pulp3_repo_setup[repo_name].get("signing_service")
+    if signing_service_name:
+        _logger.info(
+            "Signing Service %r requested for %r", signing_service_name, repo_name
+        )
+        try:
+            signing_service = SigningService.objects.get(name=signing_service_name)
+        except SigningService.DoesNotExist:
+            _logger.warning(
+                "Could not find signing-service named %r", signing_service_name
+            )
+            raise
```

Comment:
I think it's fine. I would fail since you explicitly specify in the migration plan that a certain signing service needs to be used.

I hoped that it would be a more local, plugin specific change. But since we do not provide any hooks to migrate a pulp2 repo in a custom way, the only option would be to have this check for signing service in a distributor migration. And this would mean a lot of duplicated checks for signing service existence (N repo versions and for each you use the same signing service). For that reason, I'm ok with this code to be here, though not entirely desirable.